src/urllib3/exceptions.py

```diff
@@ -154,7 +154,7 @@ class LocationValueError(ValueError, HTTPError):
     pass
 
 
-class LocationParseError(LocationValueError):
+class LocationParseError(UnicodeError, LocationValueError):
```

Comment:
I'm recommending this not be treated like a breaking change by changing the exception type on the basis of:
- `UnicodeError` clearly isn't expected to be raised from this API.
- Our test suite doesn't expect this either, nor is it documented.
- Raising `LocationParseError` is what *should* have been happening all this time, thus I view this more as a bug fix.

So we can revert adding `UnicodeError` as a subclass.

Reply:
Reverted!